lib/puppet/module/task.rb

```diff
@@ -61,6 +62,33 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.find_files(name, metadata)
+      if metadata.key?('files')
+        file_list = metadata['files'].map do |file|
+          splits = file.split("/")
```

Comment:
A couple observations for myself as I think this through.

For tasks we decided not to use the generic file URI used by Puppet (i.e. something like `puppet:///modules/apache/httpd`, `puppet:///tasks/service/init.rb`, `puppet:///plugins/stdlib/lib/puppet`) because
- unlike file source, we don't support anything besides `puppet:///` (we might change our minds, but this seems like a good starting point and doesn't exclude other options)
- listing the mount point is confusing, and we also don't want to support all mount points (again initially)
- we don't want to require other options (like recursion) to complicate the metadata format

Those choices mean we can't reuse `Puppet::FileServing::Metadata.indirection.find(source, options)` to test for the file.

Reply:
I'm not sure I follow? I mostly split on / to validate the modulename and that the next part is a valid mount path. Should I be thinking about this differently?

Reply:
Nothing to do here. I was thinking through why we couldn’t reuse other tools.